packages/common/src/utils/arrays/ArrayUtils.ts

```diff
@@ -34,6 +35,20 @@ export const ArrayUtils = {
   ): number => {
     return arr.reduce((sum, someEntity) => sum + someEntity[key], 0);
   },
+  immutableUpdateById: <T extends RecordWithId>(
```

Comment:
😁

Reply:
I'm unsure about the name! It's too long :(

Was thinking of some alternatives like `updateWithPatch`.. `patchUpdate` .. 🙄

Reply:
`patch` can be a verb too so you could shorten by dropping the update, if that's not confusing?
- `patchById`
- `patchImmutable`

Reply:
Oh, yes! nice, thanks!